src/framework/core/js/Fluid.js

```diff
@@ -1833,20 +1947,35 @@ var fluid = fluid || fluid_4_0_0; // eslint-disable-line no-redeclare
 
     /*** FLUID ERROR SYSTEM ***/
 
+    /** Upgrades a promise rejection payload (or Error) by suffixing an additional "while" reason into its "message" field
+     * @param {Object|Error} originError - A rejection payload. This should (at least) have the member `isError: true` set, as well as a String `message` holding a rejection reason.
+     * @param {String} whileMsg - A message describing the activity which led to this error
+     * @return {Object} The rejected payload formed by shallow cloning the supplied argument (if it is not an `Error`) and suffixing its `message` member
+     */
+    fluid.upgradeError = function (originError, whileMsg) {
+        var error = originError instanceof Error ? originError :
+            fluid.isPrimitive(originError) ? { message: originError} : fluid.extend({}, originError);
+        error.message = error.message + whileMsg;
+        return error;
+    };
+
     fluid.failureEvent = fluid.makeEventFirer({name: "failure event"});
 
     fluid.failureEvent.addListener(fluid.builtinFail, "fail");
     fluid.failureEvent.addListener(fluid.logFailure, "log", "before:fail");
 
     /**
+     * This function is deprecated and will be removed in the FLUID-6148 release.
```

Comment:
Does this refer to the upcoming 4.0 release?

Reply:
Yes, looks like this is a good time to do it, thanks - this was removed from Kettle a while ago which was the only known dependency